### PR TITLE
Math magicians: testing

### PR DESCRIPTION
**Pull request by ogagaoghene**
# Math Magicians React App

## Description
Testing components and arithmetic operations with Jest in React

### CODE REVIEW

File: src/tests/App.test.js [hunk 1]
```diff
@@ -0,0 +1,11 @@
...
+import renderer from 'react-test-renderer';
+import { BrowserRouter as Router } from 'react-router-dom';
+import App from '../App';
+
+it('Check if App component is rendered', () => {
+  const tree = renderer.create(
+    <Router><App /></Router>,
+  ).toJSON();
+  expect(tree).toMatchSnapshot();
+});
```
Comment (medaminedev66) [inline src/tests/App.test.js:14]: - [x] Following the project requirements, you should use Jest snapshots to test all the components. Please add a Jest snapshot test for the Calculator component.

Comment (medaminedev66) [inline src/tests/App.test.js:14]: - [x] Following the project requirements, you should use also React Testing Library to simulate user interaction. Check out this [link ](https://github.com/microverseinc/curriculum-react-redux/blob/main/math-magicians/articles/testing_react_with_react_testing_library.md)to learn how to do that.

File: src/tests/calculate.test.js [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+  it('test "-" button', () => {
+    const obj = {
+      total: '100',
+      next: '4',
+      operation: '-',
+    };
+    const result = calculate(obj, '=');
+    expect(result).toEqual({ total: '96', next: null, operation: null });
+  });
+});
```
Comment (medaminedev66) [inline src/tests/calculate.test.js]: - [ ] Following the project requirements, all the functionalities & scenarios of the `calculate.js` module should be covered using Jest.

File: src/tests/Quotes.test.js [hunk 3]
```diff
@@ -0,0 +1,16 @@
...
+describe('Snapshot testing', () => {
+  it('Check if Quotes component is rendered', () => {
+    const tree = renderer.create(<Quotes />).toJSON();
+    expect(tree).toMatchSnapshot();
+  });
+
+  it('renders quote', () => {
+    render(<Quotes />);
+    expect(screen.getByText('Mathematics is not about numbers, equations, computations, or algorithms. It is about understanding.')).toBeInTheDocument();
+  });
```
Comment (medaminedev66) [inline src/tests/Quotes.test.js]: - [x] This test fails while running `npm test`. Please try to fix that.

image

**Review by medaminedev66 — CHANGES_REQUESTED** [review]
Hi @ogagaoghene  😃, :wave:, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Highlights
- The repo looks professional
- There are no linter errors
- You have a descriptive PR
- All the functionalities & scenarios of the operate.js module were covered using Jest.

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: src/tests/App.test.js (hunk 1 above)
Comment (ogagaoghene) [reply src/tests/App.test.js:14]: This one is resolved

Comment (ogagaoghene) [reply src/tests/App.test.js:14]: Resolved

File: src/tests/Quotes.test.js (hunk 3 above)
Comment (ogagaoghene) [reply src/tests/Quotes.test.js]: image

**Review by addod19 — APPROVED** [review]
### Hi @ogagaoghene and @AngomRanjan, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

### Highlights
- [ ] Nice job fixing all the comments made by the previous code reviewer :100: 

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by ogagaoghene** [comment]
Thanks for your professional code review. I appreciate you greatly

**Pull request merged**